clang/test/Driver/linux-ld.c

```diff
@@ -1569,6 +1569,40 @@
 // RUN:        --gcc-toolchain="" \
 // RUN:        --sysroot=%S/Inputs/basic_linux_tree 2>&1 \
 // RUN:   | FileCheck --check-prefix=CHECK-NOCRTFASTMATH %s
+// Don't link crtfastmath.o with -shared
+// RUN: %clang --target=x86_64-unknown-linux -no-pie -### %s -ffast-math -shared \
+// RUN:        --gcc-toolchain="" \
```

Comment:
You can drop `--gcc-toolchain=""`. Setting CMake `GCC_INSTALL_PREFIX` leads to a warning and the option will be removed.

-no-pie/-pie/-shared are mutual exclusive. The interaction is tested by other tests. For the purpose of fastmath, remove -no-pie when -shared is used.

Reply:
Do we have trouble with this test on systems where crtfastmath.o is not provided by the gcc toolchain? I don't think we provide our own version of this, do we?

Reply:
The tests use a custom sysroot that provides a `crtfastmath.o` file, so it doesn't matter whether or not it's provided by the host toolchain.